src/python_testing/TC_MCORE_FS_1_2.py

```diff
@@ -0,0 +1,185 @@
+
+        self.step(2)
+
+        self.sync_passcode = 20202024
+        self.th_sed_dut_discriminator = 2222
+        # min commissioning timeout is 3*60 seconds, so use that even though the command said 30.
+        cmd = Clusters.AdministratorCommissioning.Commands.OpenCommissioningWindow(commissioningTimeout=3*60,
+                                                                                   PAKEPasscodeVerifier=b"+w1qZQR05Zn0bc2LDyNaDAhsrhDS5iRHPTN10+EmNx8E2OpIPC4SjWRDQVOgqcbnXdYMlpiZ168xLBqn1fx9659gGK/7f9Yc6GxpoJH8kwAUYAYyLGsYeEBt1kL6kpXjgA==",
+                                                                                   discriminator=self.th_sed_dut_discriminator,
+                                                                                   iterations=10000, salt=base64.b64encode(bytes('SaltyMcSalterson', 'utf-8')))
+        await self.send_single_cmd(cmd, endpoint=th_sed_dut_bdbi_endpoint, timedRequestTimeoutMs=5000)
+
+        logging.info("Commissioning Window open for TH_SED_DUT.")
+
+        self.step(3)
+
+        self.th_sed_dut_nodeid = 1111
+        await self.TH_server_controller.CommissionOnNetwork(nodeId=self.th_sed_dut_nodeid, setupPinCode=self.sync_passcode, filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=self.th_sed_dut_discriminator)
+        logging.info("Commissioning TH_SED_DUT complete")
+
+        self.step(4)
+        if not self.is_ci:
+            self.wait_for_use_input("Commission TH_SED_DUT onto DUT_FSA’s fabric using the manufacturer specified mechanism. (ensure Synchronization is enabled.)")
+
+        self.step(5)
+
+        th_sed_later_bdbi_endpoint = -1
+        th_sed_later_unique_id = -1
+        logging.info("Waiting for Second BridgedDeviceBasicInformation.")
+
+        while time_remaining > 0 and th_sed_later_bdbi_endpoint < 0:
+            try:
+                item = unique_id_queue.get(block=True, timeout=time_remaining)
+                client_name, endpoint, attribute, value = item['name'], item['endpoint'], item['attribute'], item['value']
+
+                # Record arrival of an expected subscription change when seen
+                if attribute == Clusters.BridgedDeviceBasicInformation.Attributes.UniqueID and endpoint != th_sed_dut_bdbi_endpoint and th_sed_later_unique_id != th_sed_dut_unique_id:
+                    th_sed_later_bdbi_endpoint = endpoint
+                    th_sed_later_unique_id = value
+
+            except queue.Empty:
+                # No error, we update timeouts and keep going
+                pass
+
+            elapsed = time.time() - start_time
+            time_remaining = timeout_delay_sec - elapsed
+
+        logging.info("Found another BDBI with UniqueID (%d) on endpoint %d." % th_sed_later_unique_id, th_sed_later_bdbi_endpoint)
+
+        self.step(6)
+
+        self.th_sed_later_discriminator = 3333
```

Comment:
as above - you can eliminate the hard coding here by using the support functions.